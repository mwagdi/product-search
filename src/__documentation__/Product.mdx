---
name: Product
menu: Components
route: /components/Product
---

import { Playground, Props } from 'docz'
import Product from '../components/Product.js'
import './../assets/scss/app.scss'

# Product

## Props
<Props of={Product} />

## Basic usage
<Playground>
  <Product
  clickable={false}
  product={{
    title: "Product Title",
    gtin: "000000",
    gender: "male",
    price: "29.99 EUR",
    sale_price: "19.99 EUR",
    image_link: "http://lp.weekday.com/app003prod?set=source[02_0514898_001_001],type[PRODUCT],device[hdpi],quality[80],ImageVersion[2019092]&call=url[file:/product/style]"
  }} />
</Playground>
