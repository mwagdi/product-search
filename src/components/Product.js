import React from 'react';

export default ({ product }) => (
    <div className="product">
        <div className="product__wrap">
            <h2 className="product__title">{product.title}</h2>
            <img src={product.image_link} alt={product.title}/>
            <ul className="product__details">
                <li className="product__details_item">
                    Gtin: {product.gtin}
                </li>
                <li className="product__details_item">
                    Gender: {product.gender}
                </li>
                <li className="product__details_item">
                    Price: {product.price}
                </li>
                <li className="product__details_item">
                    Sale Price: {product.sale_price}
                </li>
            </ul>
        </div>
    </div>
);
