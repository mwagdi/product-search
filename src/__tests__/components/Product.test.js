import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Product from '../../components/Product';

describe("Product component", () => {
    test('Renders Product component', () => {
        const product = {
            title: "Product Title",
            gtin: "gtinExample",
            gender: "female",
            price: "39.95 EUR",
            sale_price: "39.95 EUR",
            image_link: "dummy.jpg",
            additional_image_link: "*****,*****,*****"
        }
        const wrapper = shallow(<Product product={product} />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
})