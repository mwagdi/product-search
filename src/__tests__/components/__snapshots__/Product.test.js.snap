// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product component Renders Product component 1`] = `
<div
  className="product"
>
  <div
    className="product__wrap"
  >
    <h2
      className="product__title"
    >
      Product Title
    </h2>
    <img
      alt="Product Title"
      src="dummy.jpg"
    />
    <ul
      className="product__details"
    >
      <li
        className="product__details_item"
      >
        Gtin: 
        gtinExample
      </li>
      <li
        className="product__details_item"
      >
        Gender: 
        female
      </li>
      <li
        className="product__details_item"
      >
        Price: 
        39.95 EUR
      </li>
      <li
        className="product__details_item"
      >
        Sale Price: 
        39.95 EUR
      </li>
    </ul>
  </div>
</div>
`;
